internal/dump/archive.go

```diff
@@ -14,27 +16,39 @@ import (
 	"github.com/restic/restic/internal/walker"
 )
 
-// WriteTar will write the contents of the given tree, encoded as a tar to the given destination.
+// WriteArchive will write the contents of the given tree, encoded as a tar or zip to the given destination.
 // It will loop over all nodes in the tree and dump them recursively.
-func WriteTar(ctx context.Context, repo restic.Repository, tree *restic.Tree, rootPath string, dst io.Writer) error {
-	tw := tar.NewWriter(dst)
+func WriteArchive(ctx context.Context, archiveType string, repo restic.Repository, tree *restic.Tree, rootPath string, dst io.Writer) error {
```

Comment:
The zip writing logic should be disentangled from the tar code. Mixing them makes both hard to read, extend and debug.

Reply:
Agreed with @greatroar. I have [my own version](https://pastebin.com/yQez3YXB) of this PR with logic splitted to tar, zip and common. You can use it if you want.

Reply:
@DRON-666 I don't want to duplicate efforts. If you have it then I can close this PR.